File: java/oop/ContactApp/src/com/techlabs/contacts/ContactServiceProvider.java
package com.techlabs.contacts;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.Serializable;
import java.util.ArrayList;
public class ContactServiceProvider   {
	StoreData s1 = new StoreData();

	public ArrayList<Contact> contactList = new ArrayList<Contact>();
	public ContactServiceProvider() throws ClassNotFoundException, IOException{
		
		this.contactList=s1.getData();
	}
	
	
	public void addContact(String name,String number){
		this.contactList.add(new Contact(name, number));
		// serialize the list
		s1.serializeTheList(this.contactList);
		
	}
	
	public ArrayList<Contact> getNameList(){
		return this.contactList;
	}

	
	
	
	
	

}

File: dotNet/csharp/ContactApp/ContactApp/ContactServiceProvider.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ContactApp
{
    class ContactServiceProvider
    {
        StoreData s1 = new StoreData();
        public List<Contact> contactList;

        public ContactServiceProvider()
        {
            this.contactList = s1.GetData();
        }

        public void AddContact(string name,string number)
        {
            this.contactList.Add(new Contact(name,number));
            s1.SerializeTheList(this.contactList);
        }

        public List<Contact> NameList
        {
            get
            {
                return this.contactList;
            }
        }
    }
}
